nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/AddPermissionTest.java

```diff
@@ -421,6 +425,31 @@ public void shouldAddPermissionForUserWhenBlockingInheritance() throws Operation
         assertExpectedPermissions(acl, params, "existingUser1");
     }
 
+    @Test
+    public void shouldAddPermissionForUsersWhenUsingChainExtension() throws OperationException {
+        // Mock the users params
+        List<String> users = Arrays.asList("jchastain", "rdeniro", "dwashington");
+        users.forEach(user -> when(userManager.getUserModel(user)).thenReturn(new SimpleDocumentModel(user)));
+
+        DocumentModel doc = session.getDocument(new PathRef("/src"));
+        OperationContext ctx = new OperationContext(session);
+        ctx.setInput(doc);
+
+        DocumentModel documentModel = (DocumentModel) automationService.run(ctx, "testAddPermissionToUsers");
+
+        ACP acp = documentModel.getACP();
+        assertNotNull(acp);
+
+        ACL acl = acp.getACL("local");
+        assertNotNull(acl);
+
+        Map<String, Object> params = Collections.singletonMap("permission",  SecurityConstants.READ);
+        Assert.assertTrue("ACEs size must be the same then users size !", acl.getACEs().length == users.size());
```

Comment:
Do not check the exact length, the operation **adds** permissions, it does not set permissions. You can't control existing ACEs on the document. I would say...
```java
assertTrue(acl.getACEs().length >= 3);
```